### PR TITLE
Modified Ui Authentication 

### PR DESCRIPTION
**Pull request by BichraiX**
Modified UI Authentication to fill the right tables in the matrix database and have better error handling. Updated the endpoints that use Ui Auth (/add and /register) in order to comply with these changes. 
Added a README file for UI authentication to clarify usage.

### CODE REVIEW

File: packages/matrix-client-server/src/utils/setupTokens.ts [hunk 1]
```diff
@@ -59,7 +77,7 @@ export async function setupTokens(
     })
 
     await clientServer.matrixDb.insert('access_tokens', {
-      id: validRefreshTokenId1,
+      id: randomString(64),
```
[the remark this answers is not in the document]
Comment (BichraiX) [reply packages/matrix-client-server/src/utils/setupTokens.ts:80]: It is useless, I must have double inserted it and modified it afterwards to respect the unique constraint, my bad. Will remove it now

File: packages/matrix-client-server/src/utils/setupTokens.ts [hunk 2]
```diff
@@ -121,6 +124,38 @@ export async function setupTokens(
...
+      token: validToken3,
+      refresh_token_id: validRefreshTokenId3,
+      valid_until_ms: epoch() + 64000
+    })
+
+    await clientServer.matrixDb.insert('access_tokens', {
+      id: 0,
+      user_id: 'wrongUserId',
+      token: 'wrongUserAccessToken'
+    })
```
Comment (guimard) [inline packages/matrix-client-server/src/utils/setupTokens.ts:158]: What is this ? Insert always the same value ?

File: packages/matrix-client-server/src/utils/setupTokens.ts [hunk 3]
```diff
@@ -121,6 +124,38 @@ export async function setupTokens(
...
+    await clientServer.matrixDb.insert('user_threepids', {
+      user_id: '@validated:example.com',
+      medium: 'msisdn',
+      address: '0612938719',
+      validated_at: epoch(),
+      added_at: epoch()
+    })
+
+    await clientServer.matrixDb.insert('access_tokens', {
+      id: randomString(64),
```
Comment (guimard) [inline packages/matrix-client-server/src/utils/setupTokens.ts:146]: WHat is the need of this "id" field ? Since not collected

Comment (BichraiX) [reply packages/matrix-client-server/src/utils/setupTokens.ts:146]: It's the primary key of the 'access_tokens' table in the synapse DB schema so I needed to fill it even if I'm not collecting it

**Comment by guimard** [comment]
@BichraiX : merge it when test succeeds

**Review by guimard — APPROVED** [review]

**Comment by guimard** [comment]
`src/rooms/rooms.test.ts:15:8 - error TS2307: Cannot find module '../utils/setupTokens'`

**Pull request merged**